### PR TITLE
Add coils and sensors data from MDS as input

### PR DESCRIPTION
**Pull request by FilipJankyTE**
- rtgsfit reads coils and sensors from given time
 - Produces some outputs (including an input signal)
 - Add -O3 and -pthread into compilation

### CODE REVIEW

File: tests/replay_rtgsfit.c [hunk 1]
```diff
@@ -142,8 +191,16 @@ int main(int argc, char *argv[]) {
...
+  float *pcs_dec_data = (float *)calloc(pcs1_dec_len + pcs2_dec_len,
+    sizeof(float));
+  memcpy(pcs_dec_data, pcs1_dec_data, pcs1_dec_len * sizeof(float));
+  memcpy(&pcs_dec_data[pcs1_dec_len], pcs2_dec_data,
+     (pcs2_dec_len) * sizeof(float));
+  char str[100];
+  float t_step = (pcs1_dec_time[dec_N_iter - 1] - pcs1_dec_time[0]) / dec_N_iter;
+  /* t_start is referenced to plasma breakdown */
+  long int t_start_idx = (long int) ((t_start - pcs1_dec_time[0]) / t_step);
   long *t10 = calloc((size_t)N_iter, sizeof(long));
```
Comment (jbland64) [inline tests/replay_rtgsfit.c:204]: size_t vs long?

File: tests/replay_rtgsfit.c [hunk 2]
```diff
@@ -156,89 +213,38 @@ int main(int argc, char *argv[]) {
...
-  }
-  double meas[N_MEAS];
-  if (get_double_from_file(fn_meas, N_MEAS, meas) != 0) {
-    errorExit("Cannot get array for meas data.\n");
-  }
-  double coil_curr[N_COIL];
-  if (get_double_from_file(fn_coil_curr, N_COIL, coil_curr) != 0) {
-    errorExit("Cannot get array for coil current.\n");
+  for (i = 0; i < sensors_idx_num; ++i) {
+    if (sensors_idx[i] >= 0) {
```
Comment (jbland64) [inline tests/replay_rtgsfit.c:217]: when would this condition not hold?

File: tests/replay_rtgsfit.c [hunk 3]
```diff
@@ -156,89 +213,38 @@ int main(int argc, char *argv[]) {
...
+  for (i = 0; i < N_COIL; ++i) {
+    coil_curr[i] = (double)(pcs_dec_data[t_start_idx * (num_pcs1_channels + num_pcs2_channels) +
+     coil_idx[i]]);
   }
-  int mask[N_GRID];
-  if (get_int_from_file(fn_mask, N_GRID, mask) != 0) {
-    errorExit("Cannot get array for mask.\n");
+  uint32_t N_steps = 1;
+  stash_st40runner_float("pcs1_ch001", filename,
+    sizeof(float), (size_t)dec_N_iter, &pcs_dec_data[dec_N_iter * sensors_idx[29]]);
```
Comment (jbland64) [inline tests/replay_rtgsfit.c:230]: make 29 a variable of some sort

File: tests/replay_rtgsfit.c [hunk 4]
```diff
@@ -156,89 +213,38 @@ int main(int argc, char *argv[]) {
...
+  for (istep = 0; istep < N_steps;  ++istep) {
+    clock_gettime(CLOCK_MONOTONIC, &t0);
+    rtgsfit(meas, coil_curr, flux_norm, mask, psi_total, &error);
+    clock_gettime(CLOCK_MONOTONIC, &t1);
+    t10[istep] = (long)
+      ((t1.tv_sec * 1e9 + t1.tv_nsec - t0.tv_sec * 1e9 - t0.tv_nsec) / 1e3);
+      printf("time %ld\n", t10[istep]);
```
Comment (jbland64) [inline tests/replay_rtgsfit.c:237]: add unit to output time

**Review by jbland64 — APPROVED** [review]
looks good, i have made some comments, questions im not sure about?

File: tests/replay_rtgsfit.c (hunk 1 above)
Comment (FilipJankyTE) [reply tests/replay_rtgsfit.c]: size_t is returned type value from from sizeof operation

File: tests/replay_rtgsfit.c (hunk 2 above)
Comment (FilipJankyTE) [reply tests/replay_rtgsfit.c]: This in case there is no ROG. Then it will be -1

File: tests/replay_rtgsfit.c (hunk 4 above)
Comment (FilipJankyTE) [reply tests/replay_rtgsfit.c:243]: I will with the next next review

**Review by jbland64 — APPROVED** [review]
went through in person

**Pull request merged**